onnxruntime/test/contrib_ops/matmul_integer_to_float_test.cc

```diff
@@ -98,7 +98,12 @@ void TestMatMulIntegerToFloat(const std::vector<int64_t>& A_dims,
   test.SetOutputRelErr("Y", 1e-4f);
 #endif
 
-  test.Run();
+  if constexpr (std::is_same_v<OType, float>) {
+    test.Run();
```

Comment:
If I am not missing anything, then should we compare OType with MLFloat16?

Reply:
`test.Run()` tries to run the tests on all EPs, that why I check for float.
The else condition here implies the only other output type, FP16. We pass CPU EP to `excluded_provider_types` to `test.Run`

Reply:
Ohh my bad, I miss interpret the if/else condition. Thank you for explaining this!
Today this method has been called only for 2 data types `float` and `MLFloat16`, that's why it works. Otherwise, it would have meant only run for `float` data type. 
But I guess there will never be a scenario when this method will be called with any other data type except (float and float16).